client/processed_items_report/directives/ProcessedItemsReportPanel.js

```diff
@@ -55,16 +57,23 @@ export function ProcessedItemsReportPanel(config, api, session, notify, $rootSco
              * @description Sets the selected user
              */
             scope.selectUser = function(user) {
-                scope.selectedUser = user;
+                scope.selectedUser = user._id;
+                scope.selectedUsers.push(user._id);
+                scope.selectedUsersName.push(user.display_name);
             };
 
             /**
              * @ngdoc method
              * @name sdProcessedItemsReportPanel#removeUser
              * @description Removes the selected user
              */
-            scope.removeUser = function() {
-                scope.selectedUser = null;
+            scope.removeUser = function(item) {
+                for (var i = scope.selectedUsersName.length; i--;) {
+                    if (scope.selectedUsersName[i] === item) {
+                        scope.selectedUsers.splice(i, 1);
+                        scope.selectedUsersName.splice(i, 1);
```

Comment:
What if there are two users with the same display name? :) I know it's a very edge case but still..
I would make selectedUsers an array of objects. E.g.: [{_id: <user_id>, display_name: <display_name>}, ..]
This way you have only one array.